lib/backend/repository/repository2.go

```diff
@@ -198,6 +198,10 @@ func (rb *RepositoryBackend) getLayerV2(layerID string, dockerURL *types.ParsedD
 		return nil, err
 	}
 
+	if rb.username != "" && rb.password != "" {
+		req.SetBasicAuth(rb.username, rb.password)
+	}
+
```

Comment:
Yes. IIRC, in docker v1 you authenticate against the index, receive a token and then use that to query the registry. In docker v2, there's no distinction between index and registry, so you always use authentication.